lib/carconntwistprim.cc

```diff
@@ -0,0 +1,270 @@
+          else
+            tpert = t;
+          double apert = w*tpert;              // angle perturbed
+          double lpert = u*tpert;
+          if(apert > cfg.amax || lpert > cfg.lmax || lpert < 0.7*cfg.lmin)
+            continue;
+          Matrix3d gend; se2_exp(gend,Vector3d(w*tpert, u*tpert, 0));
+          Vector3d xyzend(gend(0,2),gend(1,2),0);
+          Vector3d idxd = igorg_to_car * xyzend;
+          Vector3i idxi(round(idxd(0)),round(idxd(1)),round(idxd(2)));
+
+          if((uint)idxi(1)>grid_nc-1 || (uint)idxi(0)>grid_nc-1) //Shouldn't be necessary
+            continue;
+
+          if(grid_seen(idxi(1), idxi(0))==true)
+            continue;
+          else
+            grid_seen(idxi(1), idxi(0))=true;
+
+          Vector3d xyzend_snapped = igorg_to_car.inverse()* (idxi.cast<double>()) ;
+          //cout<<"xyzend_snapped:"<<xyzend_snapped.transpose()<<endl;
+          Vector2d wtend= xy2wt2(xyzend_snapped(0), xyzend_snapped(1),u);
+          double wend = wtend(0); double tend = wtend(1);
+          Vector3d vfp(wend*tend, u*tend,0);
+          Vector3d vfn(-wend*tend, u*tend,0);
+          Vector3d vbp(wend*tend, -u*tend,0);
+          Vector3d vbn(-wend*tend, -u*tend,0);
+
+          if(abs(wend)<1e-10){
+            vss[idx_a].push_back(vfp);
+            if(!cfg.fwdonly)
+              vss[idx_a].push_back(vbp);
+          }else{
+            vss[idx_a].push_back(vfp);
+            vss[idx_a].push_back(vfn);
+            if(!cfg.fwdonly){
+              vss[idx_a].push_back(vbp);
+              vss[idx_a].push_back(vbn);
+            }
+          }
+        }
+      }
+    }
+  }else{
+
+    for(size_t idx_t=0 ; idx_t< cfg.nl;idx_t++){
+      double t = tmin*pow(del_t, idx_t);
+      for(size_t idx_w=0 ; idx_w < cfg.na; idx_w++){
+        double w = idx_w*del_w;
+        double tpert;
+        if(cfg.pert)
+          tpert = t*(1 + 0.2*(cos(40*w)-1)); // t perturbed
+        else
+          tpert = t;
+        double apert = w*tpert;              // angle perturbed
+        double lpert = u*tpert;
+        if(apert > cfg.amax || lpert > cfg.lmax || lpert < 0.7*cfg.lmin)
+          continue;
+
+        Vector3d vfp(w*tpert, u*tpert,0);
+        Vector3d vfn(-w*tpert, u*tpert,0);
+        Vector3d vbp(w*tpert, -u*tpert,0);
+        Vector3d vbn(-w*tpert, -u*tpert,0);
+        //Put the same set of primitive for all angles
+        for(int idx_a=0; idx_a < grid.gs(0);idx_a++){
+          if(abs(w)<1e-10){
+            vss[idx_a].push_back(vfp);
+            if(!cfg.fwdonly)
+              vss[idx_a].push_back(vbp);
+          }else{
+            vss[idx_a].push_back(vfp);
+            vss[idx_a].push_back(vfn);
+            if(!cfg.fwdonly){
+              vss[idx_a].push_back(vbp);
+              vss[idx_a].push_back(vbn);
+            }
+          }
+        }
+      }
+    }
+
+  }
+
+  return true;
+}
+
+bool CarConnTwistPrim::SetPrimitives(double dt, double vx, double kmax, int kseg, bool onlyfwd) {
```

Comment:
isn't this the same as the other CarConnectivity? if so why are we using two separate classes for them, can't they be combined into one with options?